File: _includes/code/java-v6/src/test/java/ManageObjectsReadAllTest.java
import io.weaviate.client6.v1.api.WeaviateClient;
import io.weaviate.client6.v1.api.collections.CollectionHandle;
import io.weaviate.client6.v1.api.collections.WeaviateObject;
import io.weaviate.client6.v1.api.collections.tenants.Tenant;
import org.junit.jupiter.api.AfterAll;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.Test;

import java.io.IOException;
import java.util.List;
import java.util.Map;

class ManageObjectsReadAllTest {

  private static WeaviateClient client;

  @BeforeAll
  public static void beforeAll() throws IOException {
    // Instantiate the client
    String openaiApiKey = System.getenv("OPENAI_API_KEY");
    client = WeaviateClient.connectToLocal(
        config -> config.setHeaders(Map.of("X-OpenAI-Api-Key", openaiApiKey)));

    // Simulate weaviate-datasets by creating and populating collections
    // Create WineReview collection
    if (client.collections.exists("WineReview")) {
      client.collections.delete("WineReview");
    }

    var wineReview = client.collections.create("WineReview");
    wineReview.data.insertMany(Map.of("title", "Review A"),
        Map.of("title", "Review B"));

    // Create WineReviewMT collection
    if (client.collections.exists("WineReviewMT")) {
      client.collections.delete("WineReviewMT");
    }
    client.collections.create("WineReviewMT",
        col -> col.multiTenancy(c -> c.autoTenantCreation(true)));
    var wineReviewMT = client.collections.use("WineReviewMT");

    // Create and populate tenants
    List<Tenant> tenants =
        List.of(Tenant.active("tenantA"), Tenant.active("tenantB"));
    wineReviewMT.tenants.create(tenants);
    wineReviewMT.withTenant("tenantA").data
        .insert(Map.of("title", "Tenant A Review 1"));
    wineReviewMT.withTenant("tenantB").data
        .insert(Map.of("title", "Tenant B Review 1"));
  }

  @AfterAll
  public static void afterAll() throws Exception {
    client.collections.delete("WineReview");
    client.collections.delete("WineReviewMT");
    client.close();
  }

  @Test
  void testReadAllProps() {
    // START ReadAllProps
    CollectionHandle<Map<String, Object>> collection =
        client.collections.use("WineReview");

    // highlight-start
    for (WeaviateObject<Map<String, Object>> item : collection.paginate()) {
      // highlight-end
      System.out.printf("%s %s\n", item.uuid(), item.properties());
    }
    // END ReadAllProps
  }

  @Test
  void testReadAllVectors() {
    // START ReadAllVectors
    CollectionHandle<Map<String, Object>> collection =
        client.collections.use("WineReview");

    for (WeaviateObject<Map<String, Object>> item : collection.paginate(
        // highlight-start
        i -> i.returnMetadata() // If using named vectors, you can specify ones to include
    // highlight-end
    )) {
      System.out.println(item.properties());
      // highlight-start
      System.out.println(item.vectors());
      // highlight-end
    }
    // END ReadAllVectors
  }

  @Test
  void testReadAllTenants() {
    // START ReadAllTenants
    CollectionHandle<Map<String, Object>> multiCollection =
        client.collections.use("WineReviewMT");

    // Get a list of tenants
    // highlight-start
    var tenants = multiCollection.tenants.get();
    // highlight-end

    // Iterate through tenants
    for (Tenant tenant : tenants) {
      // Iterate through objects within each tenant
      // highlight-start
      for (WeaviateObject<Map<String, Object>> item : multiCollection
          .withTenant(tenant.name())
          .paginate()) {
        // highlight-end
        System.out.printf("%s: %s\n", tenant.name(), item.properties());
      }
    }
    // END ReadAllTenants
  }
}

File: _includes/code/csharp/ManageObjectsReadAllTest.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Text.Json;
using System.Threading.Tasks;
using Weaviate.Client;
using Weaviate.Client.Models;
using Xunit;

namespace WeaviateProject.Tests;

public class ManageObjectsReadAllTest : IAsyncLifetime
{
    private WeaviateClient client;

    // Runs once before any tests in the class
    public async Task InitializeAsync()
    {
        client = await Connect.Local(hostname: "localhost", restPort: 8080);
        // Simulate weaviate-datasets by creating and populating collections
        // Create WineReview collection
        if (await client.Collections.Exists("WineReview"))
        {
            await client.Collections.Delete("WineReview");
        }

        var wineReview = await client.Collections.Create(
            new CollectionCreateParams { Name = "WineReview" }
        );
        await wineReview.Data.InsertMany(
            new[] { new { title = "Review A" }, new { title = "Review B" } }
        );

        // Create WineReviewMT collection
        if (await client.Collections.Exists("WineReviewMT"))
        {
            await client.Collections.Delete("WineReviewMT");
        }
        var wineReviewMT = await client.Collections.Create(
            new CollectionCreateParams
            {
                Name = "WineReviewMT",
                MultiTenancyConfig = new MultiTenancyConfig
                {
                    Enabled = true,
                    AutoTenantCreation = true,
                },
            }
        );

        // Create and populate tenants
        await wineReviewMT.Tenants.Create(["tenantA", "tenantB"]);
        await wineReviewMT.WithTenant("tenantA").Data.Insert(new { title = "Tenant A Review 1" });
        await wineReviewMT.WithTenant("tenantB").Data.Insert(new { title = "Tenant B Review 1" });
    }

    // Runs once after all tests in the class
    public async Task DisposeAsync()
    {
        await client.Collections.Delete("WineReview");
        await client.Collections.Delete("WineReviewMT");
    }

    [Fact]
    public async Task TestReadAllProps()
    {
        // START ReadAllProps
        var collection = client.Collections.Use("WineReview");

        // highlight-start
        await foreach (var item in collection.Iterator())
        {
            // highlight-end
            Console.WriteLine($"{item.UUID} {JsonSerializer.Serialize(item.Properties)}");
        }
        // END ReadAllProps
    }

    [Fact]
    public async Task TestReadAllVectors()
    {
        // START ReadAllVectors
        var collection = client.Collections.Use("WineReview");

        await foreach (
            var item in collection.Iterator(
                // highlight-start
                includeVectors: true // If using named vectors, you can specify ones to include
            )
        )
        // highlight-end
        {
            Console.WriteLine(JsonSerializer.Serialize(item.Properties));
            // highlight-start
            Console.WriteLine(JsonSerializer.Serialize(item.Vectors));
            // highlight-end
        }
        // END ReadAllVectors
    }

    [Fact]
    public async Task TestReadAllTenants()
    {
        // START ReadAllTenants
        var multiCollection = client.Collections.Use("WineReviewMT");

        // Get a list of tenants
        // highlight-start
        var tenants = await multiCollection.Tenants.List();
        // highlight-end

        // Iterate through tenants
        foreach (var tenant in tenants)
        {
            // Iterate through objects within each tenant
            // highlight-start
            var tenantCollection = multiCollection.WithTenant(tenant.Name);
            await foreach (var item in tenantCollection.Iterator())
            {
                // highlight-end
                Console.WriteLine($"{tenant.Name}: {JsonSerializer.Serialize(item.Properties)}");
            }
        }
        // END ReadAllTenants
    }
}
